### ISSUE TITLE
[Bug] Tranformer Seq2Seq Have Wrong Inputs!

### ISSUE THREAD

**Issue by bot66** [bentrevett/pytorch-seq2seq#182]
**File**:[6 - Attention is All You Need](https://github.com/bentrevett/pytorch-seq2seq/blob/master/6%20-%20Attention%20is%20All%20You%20Need.ipynb)

**Problem**:

At the train step, input trg is:
`output, _ = model(src, trg[:,:-1])`
the slice indexing is for drop \<eos\> token right?

> As we want our model to predict the \<eos\> token but not have it be an input into our model we simply slice the \<eos\> token off the end of the sequence. Thus

But the trg is padded, \<eos\> token isn't the last element, during training, the \<eos\> token still trained with attentions.

The right way is mask the \<eos\> token when make trg_mask, but in the code trg_mask only mask on \<pad\> token.

**Comment by bot66** [comment]
Thanks for you replying ! Ok i get it, when calculate loss the <eos> token predict logit  loss will be ignored since the corresponding target is <pad>.

What i do is something like replace the <eos> token to <pad> token when trg input decoder transformer🤣
